spytial/dataclass_widget.py

```diff
@@ -109,13 +121,62 @@ def _setup_widget(self):
                                 console.error('Jupyter kernel not available');
                             }}
                         }}
+                        
+                        // Handle value request response from iframe
+                        else if (event.data && event.data.type === 'spytial-value-response' && event.data.widgetId === '{self._widget_id}') {{
+                            console.log('Received value response:', event.data);
+                            
+                            // Execute Python to set response data
+                            var code = `
+# Handle value request response
+try:
+    import json
+    from spytial.dataclassbuilder import json_to_dataclass
+    
+    # Get widget from global registry
+    widget_id = '{self._widget_id}'
+    if '_spytial_widgets' in globals() and widget_id in _spytial_widgets:
+        widget = _spytial_widgets[widget_id]
+        data = ${{JSON.stringify(event.data.data)}}
+        
+        # Set response data and clear waiting flag
+        widget._response_data = json_to_dataclass(data, widget.dataclass_type)
```

Comment:
I understand one timeout, but where are there timeouts in the dataclass builder?